### PR TITLE
collection wise rbac

### PR DESCRIPTION
**Pull request by notshivansh**

### CODE REVIEW

File: apps/dashboard/src/main/resources/struts.xml [hunk 1]
```diff
@@ -2467,6 +2539,49 @@
             </result>
        </action>
 
+        <action name="api/updateUserCollections" class="com.akto.action.ApiCollectionsAction" method="updateUserCollections">
+            <interceptor-ref name="json"/>
```
Comment (Ark2307) [inline apps/dashboard/src/main/resources/struts.xml:2543]: put behind stigg as well, RBAC_FEATURE extension

File: apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/settings/users/Users.jsx [hunk 2]
```diff
@@ -18,8 +22,29 @@ const Users = () => {
...
+    const handleSelectedItems = (id, items) => {
+        setSelectedItems(prevSelectedItems => ({
+            ...prevSelectedItems,
+            [id]: items
+        }));
+    }
+
+
+    if (!stiggFeatures || Object.keys(stiggFeatures).length === 0) {
+        rbacAccess = true
```
Comment (Ark2307) [inline apps/dashboard/web/polaris_web/web/src/apps/dashboard/pages/settings/users/Users.jsx:44]: this function can be replaced with the same variable rbacAccess?

File: apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java [hunk 3]
```diff
@@ -260,6 +266,27 @@ public String createCollection() {
...
+                    Filters.and(
+                            Filters.eq(RBAC.USER_ID, userId),
+                            Filters.eq(RBAC.ACCOUNT_ID, accountId),
+                            Filters.ne(RBAC.ROLE, RBAC.Role.ADMIN.getName())
+                    ),
+                    Updates.addToSet(RBAC.API_COLLECTIONS_ID, apiCollection.getId()),
+                    new UpdateOptions().upsert(false)
+            );
+    
+            UsersCollectionsList.deleteCollectionIdsFromCache(userId, accountId);
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java:286]: can we add this in delete collections as well, for maintaining atomicity

File: apps/dashboard/src/main/java/com/akto/utils/Utils.java [hunk 4]
```diff
@@ -607,13 +602,13 @@ public static void deleteApis(List<Key> toBeDeleted) {
...
-        AccountsContextDao.deleteApisPerDao(toBeDeleted, SampleDataDao.instance, id);
-        AccountsContextDao.deleteApisPerDao(toBeDeleted, TrafficInfoDao.instance, id);
-        AccountsContextDao.deleteApisPerDao(toBeDeleted, SensitiveSampleDataDao.instance, id);
-        AccountsContextDao.deleteApisPerDao(toBeDeleted, SensitiveParamInfoDao.instance, "");
-        AccountsContextDao.deleteApisPerDao(toBeDeleted, FilterSampleDataDao.instance, id);
+        AccountsContextDaoWithRbac.deleteApisPerDao(toBeDeleted, SingleTypeInfoDao.instance, "");
+        AccountsContextDaoWithRbac.deleteApisPerDao(toBeDeleted, ApiInfoDao.instance, id);
+        AccountsContextDaoWithRbac.deleteApisPerDao(toBeDeleted, SampleDataDao.instance, id);
+        AccountsContextDaoWithRbac.deleteApisPerDao(toBeDeleted, TrafficInfoDao.instance, id);
+        AccountsContextDaoWithRbac.deleteApisPerDao(toBeDeleted, SensitiveSampleDataDao.instance, id);
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/utils/Utils.java:609]: we dont need it here. dont stop writes

File: apps/dashboard/src/main/java/com/akto/listener/KafkaListener.java [hunk 5]
```diff
@@ -21,6 +21,12 @@ public void contextInitialized(javax.servlet.ServletContextEvent sce) {
             loggerMaker.infoAndAddToDb("is_kubernetes: true", LogDb.DASHBOARD);
             return;
         }
+
+        if (DashboardMode.isLocalDeployment() && !DashboardMode.isSaasDeployment()) {
+            loggerMaker.infoAndAddToDb("local: true", LogDb.DASHBOARD);
```
Comment (Ark2307) [inline apps/dashboard/src/main/java/com/akto/listener/KafkaListener.java:26]: why this?

File: libs/dao/src/main/java/com/akto/dao/AccountsContextDaoWithRbac.java [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+    protected Bson addRbacFilter(Bson originalQuery) {
+        try {
+            if (Context.userId.get() != null && Context.accountId.get() != null) {
+                List<Integer> apiCollectionIds = UsersCollectionsList.getCollectionsIdForUser(Context.userId.get(),
+                        Context.accountId.get());
+                if (apiCollectionIds != null) {
+                    List<Bson> filters = new ArrayList<>();
+                    filters.add(Filters.and(Filters.exists(getFilterKeyString()),
+                            Filters.in(getFilterKeyString(), apiCollectionIds)));
```
Comment (Ark2307) [inline libs/dao/src/main/java/com/akto/dao/AccountsContextDaoWithRbac.java:50]: do we need in both?

File: libs/dao/src/main/java/com/akto/dao/MCollection.java [hunk 7]
```diff
@@ -91,18 +91,6 @@ public<V, U> List<T> findAll(String key1, V value1, String key2, U value2) {
         return this.findAll(and(eq(key1, value1), eq(key2, value2)));
     }
 
-    public<V> List<T> findAll(String key, Collection<V> values) {
-        MongoCursor<T> cursor = this.getMCollection().find(in(key, values)).cursor();
-
-        ArrayList<T> ret = new ArrayList<T>();
-
-        while(cursor.hasNext()) {
```
Comment (Ark2307) [inline libs/dao/src/main/java/com/akto/dao/MCollection.java:99]: check for instances

File: apps/dashboard/src/main/java/com/akto/listener/KafkaListener.java (hunk 5 above)
Comment (notshivansh) [reply apps/dashboard/src/main/java/com/akto/listener/KafkaListener.java:26]: local deployment does not have kafka dependency. so skipping connecting to kafka in local deployment. Seperate bug fix, irrelevant to current feature.

File: libs/dao/src/main/java/com/akto/dao/AccountsContextDaoWithRbac.java (hunk 6 above)
Comment (notshivansh) [reply libs/dao/src/main/java/com/akto/dao/AccountsContextDaoWithRbac.java:50]: precautionary.

File: libs/dao/src/main/java/com/akto/dao/MCollection.java (hunk 7 above)
Comment (notshivansh) [reply libs/dao/src/main/java/com/akto/dao/MCollection.java:99]: yup, compiles, so should not be a problem

File: apps/dashboard/src/main/resources/struts.xml (hunk 1 above)
Comment (notshivansh) [reply apps/dashboard/src/main/resources/struts.xml:2543]: cool

File: apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java (hunk 3 above)
Comment (notshivansh) [reply apps/dashboard/src/main/java/com/akto/action/ApiCollectionsAction.java:286]: cool

**Review by Ark2307 — APPROVED** [review]

**Pull request merged**